pkg/cosign/tuf/client.go

```diff
@@ -558,25 +670,75 @@ func (e *embedded) Get(p string) ([]byte, error) {
 	return b, nil
 }
 
-type file struct {
-	base string
-	setImpl
+func (e *embeddedWrapper) Set(name string, b []byte) error {
+	// An embedded targetImpl needs to set a file, copy files to cache.
+	if e.embedded {
+		ed, ok := GetEmbedded().(fs.ReadDirFS)
+		if !ok {
+			return errors.New("fs.ReadFileFS unimplemented for embedded repo")
+		}
+		entries, err := ed.ReadDir(filepath.FromSlash(filepath.Join("repository", "targets")))
+		if err != nil {
+			return err
+		}
+		rd, ok := GetEmbedded().(fs.ReadFileFS)
+		if !ok {
+			return errors.New("fs.ReadFileFS unimplemented for embedded repo")
+		}
+		for _, entry := range entries {
+			b, err := rd.ReadFile(filepath.FromSlash(filepath.Join("repository", "targets", entry.Name())))
+			if err != nil {
+				return fmt.Errorf("reading embedded file: %w", err)
+			}
+			if err := e.writeable.Set(entry.Name(), b); err != nil {
+				return fmt.Errorf("setting embedded file: %w", err)
+			}
+		}
+		e.embedded = false
```

Comment:
IMO this makes more sense next to the `e.writeable.Set` line (i.e., you do it unconditionally). And then a comment like "now that we've updated `writeable`, we need to check it when we get targets"